File: clients/redis/python/memorix_client_redis/example_schema_generated.py
# flake8: noqa
import typing
import os

if typing.TYPE_CHECKING:
    from dataclasses import dataclass
else:
    from memorix_client_redis import dataclass

from enum import Enum
from memorix_client_redis import (
    MemorixBase,
    MemorixCacheAll,
    MemorixPubSubAll,
    MemorixTaskAll,
    Value,
)


class Animal(str, Enum):
    dog = "dog"
    cat = "cat"
    person = "person"


@dataclass
class InlineTypeUser(object):
    name: str
    age: typing.Optional[int]


User: typing.TypeAlias = InlineTypeUser


class MemorixCache(MemorixCacheAll.Base):
    def __init__(self, api: MemorixBase) -> None:
        super().__init__(api=api)

        self.favoriteAnimal = MemorixCacheAll.ItemTTTT[str, "Animal"](
            api=api,
            id="favoriteAnimal",
            payload_class=Animal,
        )
        self.user = MemorixCacheAll.ItemTTTT[str, "User"](
            api=api,
            id="user",
            payload_class=User,
        )


class MemorixPubSub(MemorixPubSubAll.Base):
    def __init__(self, api: MemorixBase) -> None:
        super().__init__(api=api)

        self.message = MemorixPubSubAll.ItemTTNoKey[str](
            api=api,
            id="message",
            payload_class=str,
        )


class MemorixTask(MemorixTaskAll.Base):
    def __init__(self, api: MemorixBase) -> None:
        super().__init__(api=api)

        self.runAlgo = MemorixTaskAll.ItemTTTTNoKey[str](
            api=api,
            id="runAlgo",
            payload_class=str,
        )


class Memorix(MemorixBase):
    def __init__(self) -> None:
        super().__init__(redis_url=Value.from_env_variable("REDIS_URL"))

        self._namespace_name_tree = []
        self.cache = MemorixCache(self)
        self.pubsub = MemorixPubSub(self)
        self.task = MemorixTask(self)

File: clients/redis/rust/src/example_schema_generated.rs
extern crate memorix_redis;
extern crate serde;
extern crate serde_json;

#[allow(non_camel_case_types, clippy::upper_case_acronyms)]
#[derive(serde::Serialize, serde::Deserialize, PartialEq)]
pub enum Animal {
    dog,
    cat,
    person,
}

#[derive(serde::Serialize, serde::Deserialize)]
pub struct User {
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub age: Option<i32>,
}

// #[derive(memorix_redis::)]
// struct FavoriteAnimalCacheItem {
//     payload: Animal
// }

// impl MemorixCacheItemNoKeyBase for FavoriteAnimalCacheItem {
//     fn getId() {
//         "favoriteAnimal"
//     }
// }

// #[derive(MemorixCacheBase)]
struct MemorixCache {
    favorite_animal: memorix_redis::MemorixCacheItemNoKey<Animal>,
}

impl MemorixCache {
    fn new() -> MemorixCache {
        MemorixCache {
            favorite_animal: memorix_redis::MemorixCacheItemNoKey<Animal>::new
        }
    }
}

// class MemorixCache(MemorixCacheBase):
//     def __init__(self, api: MemorixBase) -> None:
//         super().__init__(api=api)

//         self.favoriteAnimal = MemorixCacheItem[String, Animal](
//             api=api,
//             id="favoriteAnimal",
//             payload_class=Animal,
//         )
//         self.user = MemorixCacheItem[String, User](
//             api=api,
//             id="user",
//             payload_class=User,
//         )

// class MemorixPubSub(MemorixPubSubBase):
//     def __init__(self, api: MemorixBase) -> None:
//         super().__init__(api=api)

//         self.message = MemorixPubSubItemNoKey[String](
//             api=api,
//             id="message",
//             payload_class=String,
//         )

// class MemorixTask(MemorixTaskBase):
//     def __init__(self, api: MemorixBase) -> None:
//         super().__init__(api=api)

//         self.runAlgo = MemorixTaskItemNoKey[String, Animal](
//             api=api,
//             id="runAlgo",
//             payload_class=String,
//             returns_class=Animal,
//         )

// class Memorix(MemorixBase):
//     def __init__(
//         self,
//         redis_url: str,
//         ref: typing.Optional[MemorixBase] = None,
//     ) -> None:
//         super().__init__(redis_url=redis_url, ref=ref)

//         self._namespace_name_tree: typing.List[str] = []

//         self.cache = MemorixCache(self)
//         self.pubsub = MemorixPubSub(self)
//         self.task = MemorixTask(self)
